SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs

```diff
@@ -166,43 +167,110 @@ protected virtual async Task ImportRelationshipsAsync(IDictionary<string, BasicR
             Logger.LogInformation("Total Relationships to create: {relationshipsCount}", relationships.Count);
             var relationshipCreateSuccesses = 0;
             var relationshipCreateFailures = 0;
+            var relationshipUpdateSuccesses = 0;
+            var relationshipUpdateFailures = 0;
+            var relationshipUpdateSkips = 0;
             var failedRelationships = new ConcurrentDictionary<string, BasicRelationship>();
 
             // We need to shuffle this list to try to reduce the chances of going over the limit for the number of updates per twin per second
             var relationshipList = new List<BasicRelationship>(relationships.Values).Shuffle();
-            var relationshipsMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsync", Metrics.RelationshipTypeDimensionName);
-            var throttledMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsyncThrottle");
-            var failedMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsyncFailed", Metrics.RelationshipTypeDimensionName);
+
+            var createRelationshipSucceededMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsync", Metrics.RelationshipTypeDimensionName);
+            var createRelationshipThrottledMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsyncThrottled");
+            var createRelationshipFailedMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsyncFailed", Metrics.RelationshipTypeDimensionName);
```

Comment:
I know this is an already existing pattern, but should we swap the status to be a dimension?

```
var createRelationshipMetricIdentifier = new MetricIdentifier(Metrics.DefaultNamespace, "CreateOrReplaceRelationshipAsync", Metrics.RelationshipTypeDimensionName, "Status");

TelemetryClient.GetMetric(createRelationshipMetricIdentifier).TrackValue(1, relationship.Name, "Success");
TelemetryClient.GetMetric(createRelationshipMetricIdentifier).TrackValue(1, relationship.Name, "Throttled");
TelemetryClient.GetMetric(createRelationshipMetricIdentifier).TrackValue(1, relationship.Name, "Failed");
```

Might take some more playing with the dashboard, I dont know how easy it is to surface two dimensions

Reply:
Could also go a step further with 

TelemetryClient.GetMetric(relationshipMetricIdentifier).TrackValue(1, relationship.Name, "Create", "Skipped");

Again, Geneva, easy-peasy, AzurePortal/AppInsights, not so confident